job-tutorial/README.md

```diff
@@ -88,41 +88,43 @@ A json file describe detailed configuration required for a job submission. The d
+| Field Name                       | Schema                     | Description                              |
+| :------------------------------- | :------------------------- | :--------------------------------------- |
+| `jobName`                        | String in `^[A-Za-z0-9\-._~]+$` format, required | Name for the job, need to be unique |
+| `image`                          | String, required           | URL pointing to the Docker image for all tasks in the job |
+| `authFile`                       | String, optional, HDFS URI | Docker registry authentication file existing on HDFS |
+| `dataDir`                        | String, optional, HDFS URI | Data directory existing on HDFS          |
+| `outputDir`                      | String, optional, HDFS URI | Output directory on HDFS, `$PAI_DEFAULT_FS_URI/Output/$jobName` will be used if not specified |
+| `codeDir`                        | String, optional, HDFS URI | Code directory existing on HDFS          |
+| `virtualCluster`                 | String, optional           | The virtual cluster job runs on. If omitted, the job will run on `default` virtual cluster    |
+| `taskRoles`                      | List, required             | List of `taskRole`, one task role at least |
+| `taskRole.name`                  | String in `^[A-Za-z0-9._~]+$` format, required | Name for the task role, need to be unique with other roles |
+| `taskRole.taskNumber`            | Integer, required          | Number of tasks for the task role, no less than 1 |
+| `taskRole.cpuNumber`             | Integer, required          | CPU number for one task in the task role, no less than 1 |
+| `taskRole.memoryMB`              | Integer, required          | Memory for one task in the task role, no less than 100 |
+| `taskRole.shmMB`                 | Integer, optional          | Shared memory for one task in the task role, no more than memory size. The default value is 64MB |
+| `taskRole.gpuNumber`             | Integer, required          | GPU number for one task in the task role, no less than 0 |
+| `taskRole.portList`              | List, optional             | List of `portType` to use                |
+| `taskRole.portType.label`        | String in `^[A-Za-z0-9._~]+$` format, required | Label name for the port type |
+| `taskRole.portType.beginAt`      | Integer, required          | The port to begin with in the port type, 0 for random selection |
+| `taskRole.portType.portNumber`   | Integer, required          | Number of ports for the specific type    |
+| `taskRole.command`               | String, required           | Executable command for tasks in the task role, can not be empty |
+| `taskRole.minFailedTaskCount`    | Integer, optional          | Number of failed tasks to kill the entire job, no less than 1   |
+| `taskRole.minSucceededTaskCount` | Integer, optional          | Number of succeeded tasks to kill the entire job, no less than 1 |
```

Comment:
Better to link to launcher user manual for details, such as 
[https://github.com/Microsoft/pai/blob/master/frameworklauncher/doc/USERMANUAL.md#ApplicationCompletionPolicy](https://github.com/Microsoft/pai/blob/master/frameworklauncher/doc/USERMANUAL.md#ApplicationCompletionPolicy)

And at the bottom link to the entire doc:
[https://github.com/Microsoft/pai/blob/master/frameworklauncher/doc/USERMANUAL.md](https://github.com/Microsoft/pai/blob/master/frameworklauncher/doc/USERMANUAL.md)

Reply:
updated